models/classes/media/LocalItemSource.php

```diff
@@ -61,15 +61,16 @@ public function getItem()
      * Get a array representing the tree of directory
      *
      * @see \oat\tao\model\media\MediaBrowser::getDirectory
-     * @param string $parentLink
-     * @param array $acceptableMime
-     * @param int $depth
-     * @return array
-     * @throws \FileNotFoundException
      * @throws \tao_models_classes_FileNotFoundException
      * @throws common_exception_Error
      */
-    public function getDirectory($parentLink = '', $acceptableMime = [], $depth = 1)
+    public function getDirectory(
+        string $parentLink = '',
+        array $acceptableMime = [],
+        int $depth = 1,
+        int $limit = 0,
+        int $offset = 0
+    ): array
```

Comment:
Proper pagination support should be introduced after migration to `flysystem` 2.0 @see https://flysystem.thephpleague.com/v2/docs/usage/directory-listings/ in order to have reliable filtering